### ISSUE TITLE
Lots of errors when I start jedi-server for the first time

### ISSUE THREAD

**Issue by nayefc** [tkf/emacs-jedi#214]
I just installed jedi on Emacs. When I run jedi:install-server, I get:

```
Running: pip install --upgrade /Users/nayefcopty/dotfiles/emacs/.emacs.d/.cask/24.4.1/elpa/jedi-core-20150318.1613/...Done
deferred error : (error "Process [/Users/nayefcopty/.emacs.d/.python-environments/default/bin/pip] exited abnormally : /usr/local/Cellar/emacs/24.4/Emacs.app/Contents/MacOS/Emacs: /Users/nayefcopty/.emacs.d/.python-environments/default/bin/pip: No such file or directory
")
```

Any idea what that is? I can verify that `/Users/nayefcopty/.emacs.d/.python-environments/default/bin/pip` exists.

Thanks

**Comment by syohex** [comment]
I cannot reproduce this issue.

- Which python version did you use ?
- Did you install `python-environment.el` and `virtualenv` ?
- Did you get such error after first time ?

**Comment by nayefc** [comment]
Python 2.7.3.

I do have `python-environment.el` installed via MELPA.

What do you mean by `virtualenv`? Just the regular MELPA `virtualenv`? I use pyenv-mode instead.

Does Jedi only run when a virtualenv is activated in Emacs? I am having issues with pyenv-mode which I am fixing at the moment. Is it because of that? Does Jedi only run in a virtualenv, such that I need to run install-server only once a virtualenv is activated?

Thank you.

**Comment by nayefc** [comment]
Python 2.7.3.

I do have `python-environment.el` installed via MELPA.

What do you mean by `virtualenv`? Just the regular MELPA `virtualenv`? I use pyenv-mode instead. I also tried installing `virtualenv` and then run `jedi:install-server` but I got the same error.

Does Jedi require an active `virtualenv`?

I reinstalled Jedi, activated virtualenv using `pyenv-mode` and ran `jedi:install-server` and got:

```
Running: pip install --upgrade /Users/nayefcopty/dotfiles/emacs/.emacs.d/.cask/24.4.1/elpa/jedi-core-20150318.1613/...Done
deferred error : (error "Process [/Users/nayefcopty/.emacs.d/.python-environments/default/bin/pip] exited abnormally : /usr/local/Cellar/emacs/24.4/Emacs.app/Contents/MacOS/Emacs: /Users/nayefcopty/.emacs.d/.python-environments/default/bin/pip: No such file or directory
")
Running: pip install --upgrade /Users/nayefcopty/dotfiles/emacs/.emacs.d/.cask/24.4.1/elpa/jedi-core-20150318.1613/...Done
deferred error : (error "Process [/Users/nayefcopty/.emacs.d/.python-environments/default/bin/pip] exited abnormally : /usr/local/Cellar/emacs/24.4/Emacs.app/Contents/MacOS/Emacs: /Users/nayefcopty/.emacs.d/.python-environments/default/bin/pip: No such file or directory
")
Running: pip install --upgrade /Users/nayefcopty/dotfiles/emacs/.emacs.d/.cask/24.4.1/elpa/jedi-core-20150318.1613/...Done
deferred error : (error "Process [/Users/nayefcopty/.emacs.d/.python-environments/default/bin/pip] exited abnormally : /usr/local/Cellar/emacs/24.4/Emacs.app/Contents/MacOS/Emacs: /Users/nayefcopty/.emacs.d/.python-environments/default/bin/pip: No such file or directory
```

It is not finding pip apparently. What does Jedi require in terms of packages and virtualenv?

**Comment by syohex** [comment]
> What do you mean by virtualenv? Just the regular MELPA virtualenv?

Sorry. It is [virtualenv](https://virtualenv.pypa.io/en/latest/) command, not Emacs lisp `virtualenv` package.

**Comment by nayefc** [comment]
Yes I do have virtualenv (using pyenv). I ended up installing jediepcserver manually, which worked. I will probably end up just stop using Jedi since it's not that easy to get working.

**Comment by syohex** [comment]
I cannot reproduce this issue on my Mac OSX Yosemite(Python 2.7.6).
- Install jedi.el from MELPA
- Open python file
- M-x jedi:install-server

then there are no problems. Hmm.

**Comment by madhavan020985** [comment]
jedi install via melpa possibly might have put an old version of jedi. The reason, i am saying this is, M-x jedi:show-jedi-version buffer shows, emacs-jedi-version: "0.2.1" & python-jedi version: 0.8.1-final0. With this, my jedi doesn't lookup for pip exe inside .emacs.d folder

**Comment by nayefc** [comment]
The `jedi:show-version-info` shows `0.2.1` for `emacs-jedi`, and `0.8.1` for `jedi` and `0.0.5` for `epc`. Seems like these are all the latest.

**Comment by pcompassion** [comment]
I had this problem when I had .emacs.d/.python-enviornment which probably I installed manually not using melpa. delete the directory and reinstall python-environment, and it works fine.
